book/src/01_getting_started/03_test.md

```diff
@@ -0,0 +1,42 @@
+# Test
+
+Functionality in EDR is tested in two way:
+
+1. Rust unit & integration tests
+2. End-to-end (E2E) tests of EDR in Hardhat
+
+As EDR matures, we will gradually be moving over Hardhat E2E tests to granular unit & integration tests in EDR.
+
+## EDR
+
+Part of EDR's test suite requires a working internet connection. Those tests are marked with the `test-remote` feature flag.
+
+To run all tests, including remote tests, execute:
+
+```bash
+cargo t --all-features
+```
+
+To only run local tests, execute:
+
+```bash
+cargo t --features bench-once,serde,std,tracing
+```
+
+The `bench-once` feature flag is used to ensure that benchmarks only run one iteration, to avoid taking too long.
+
+## Hardhat
+
+To validate that the port of Hardhat Node to EDR did not break any functionality, we implemented the EDR integration alongside the existing TypeScript code.
+Each system in hidden behind an interface that allows us to execute the original Hardhat implementation, EDR, or a dual-mode adapter that executes both implementations side-by-side and asserts that outputs are equal.
+
+To switch modes, set the `HARDHAT_EXPERIMENTAL_VM_MODE` environment variable to one of: `ethereumjs`, `rethnet`, or `dual` (default). E.g.:
```

Comment:
consider placing after `rethnet` something like "(representing EDR)"

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4262